### DISCUSSION TITLE
Multiple Sales Document Item line

### DISCUSSION THREAD

**Discussion by ROLLS123** [dbosoft/YaNco#183] [Q&A] [answered]
base in this below codes  only 1 item line support there's a way to  set table to multiple  Document Item line https://github.com/dbosoft/YaNco/blob/0bf4705b4992d67ac19bf54aa9058f6664911e39/samples/netcore3.1/CreateSalesOrder/CreateSimpleSalesDocument.cs#L43-L98

  //we only support documents with one item line 
                 .SetTable("SALES_ITEMS_IN", OneLine, (s, _) => s 
                     .SetField("MATERIAL", ProductId) 
                 ) 
                 .SetTable("SALES_ITEMS_INX", OneLine, (s, _) => s 
                     .SetField("MATERIAL", "X") 
                 ) 
                 .SetTable("SALES_PARTNERS", OneLine, (s, _) => s 
                     .SetField("PARTN_ROLE", "AG") 
                     .SetField("PARTN_NUMB", CustomerNo) 
                 ) 

Thank you,

**Comment by fw2568** [comment, marked as answer]
Hi,
sure. 
The second argument of SetTable (inputList) is a `IEnumerable<TInput>`. `TInput` could be any type. 
SetTable will create a row in the function table for each entry in inputList. 

So you simple fill a array or list or any other enumerable with the data you would like to be mapped to the function table.

For example: 

``` csharp
 // partner data
  var partners = new[]
  {
      new { Role = "AG", No = "4711" },
      new { Role = "WE", No = "4712" }
  };

// later in function definition: 
....
    .SetTable("SALES_PARTNERS", partners, (s, partner) => s  //this function will be called for each entry in partners
        .SetField("PARTN_ROLE", partner.Role)
        .SetField("PARTN_NUMB", partner.No)
    )
```

See also [readme ](https://github.com/dbosoft/YaNco#using) (Input mapping).

Best Regards,
Frank

**Reply by ROLLS123** [reply]
thanks @fw2568  this is working from my side
